app/controllers/declaration/TransportContainerController.scala

```diff
@@ -44,33 +45,52 @@ class TransportContainerController @Inject()(
   errorHandler: ErrorHandler,
   override val exportsCacheService: ExportsCacheService,
   mcc: MessagesControllerComponents,
+  addFirstPage: transport_container_add_first,
   addPage: transport_container_add,
   summaryPage: transport_container_summary,
   removePage: transport_container_remove
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with ModelCacheable {
 
   def displayAddContainer(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
-    Ok(addPage(mode, TransportInformationContainer.form()))
+    if (request.cacheModel.hasContainers) {
+      Ok(addPage(mode, ContainerAdd.form()))
+    } else {
+      Ok(addFirstPage(mode, ContainerYesNo.form()))
+    }
   }
 
   def submitAddContainer(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    val boundForm = TransportInformationContainer.form().bindFromRequest()
-    saveContainer(mode, boundForm, maxNumberOfItems, request.cacheModel.containers)
+    if (request.cacheModel.hasContainers) {
+      val boundForm = ContainerAdd.form().bindFromRequest()
+      saveAdditionalContainer(mode, boundForm, maxNumberOfItems, request.cacheModel.containers)
+    } else {
+
+      ContainerYesNo.form
+        .bindFromRequest()
+        .fold(
+          (formWithErrors: Form[ContainerYesNo]) => Future.successful(BadRequest(addFirstPage(mode, formWithErrors))),
+          validForm =>
+            validForm.id match {
+              case Some(containerId) => saveFirstContainer(mode, containerId)
+              case _                 => successful(navigator.continueTo(routes.SummaryController.displayPage(Mode.Normal)))
```

Comment:
I prefer to have whole `Future.successful` not just `successful`.

Comment to whole file

Reply:
Done